deployments/gcp-uscentral1b/Makefile

```diff
@@ -0,0 +1,72 @@
+PROJECT?=pangeo-181919
+NAMESPACE?=staging
+ZONE?=us-central1-b
+CLUSTER?=pangeo-uscentral1b
+
+# Creation of the Kubernetes Cluster for Google
+#
+# Notes
+# -----
+# The pangeo-rbac stuff is untested.
+#
+# References
+# ----------
+# https://cloud.google.com/kubernetes-engine/docs/how-to/node-auto-provisioning
+# https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity
+
+# ----------------------------------------------------------------------------
+# Kubernetes Cluster
+#
+#
+# TODO: enable nodepool auto-provisioning
+		# --enable-autoprovisioning --min-cpu 1 --min-memory 1 --max-cpu 1000 --max-memory 5200 \
+		# --autoprovisioning-service-account=pangeo \
+
+test:
+	@echo $(CLUSTER) $(WORKER_POOL)
+
+up:
+	echo "[Creating cluster pool in $(ZONE)]"
+	gcloud container clusters create $(CLUSTER) \
+		--num-nodes=1 \
+		--zone=$(ZONE) \
+		--labels=cluster=$(CLUSTER) \
+		--cluster-version=latest \
+		--no-enable-ip-alias \
+		--no-enable-legacy-authorization \
+		--enable-autoupgrade --enable-autorepair --max-surge-upgrade=1 \
+		--machine-type=n1-standard-1 \
+		--enable-autoscaling --min-nodes=1 --max-nodes=2 \
+		--node-labels="hub.jupyter.org/nodepurpose=core"
+		--enable-autoprovisioning --autoprovisioning-config-file autoprovisioning.json \
+		--autoprovisioning-service-account=pangeo \
+		--enable-vertical-pod-autoscaling \
+		--workload-metadata=GKE_METADATA \
+		--workload-pool=$(PROJECT).svc.id.goog \
+		--enable-stackdriver-kubernetes
```

Comment:
also pinging @yuvipanda and @consideRatio for input on the standard/best-practice argument set for security and telemetry.

Reply:
Creating a "private cluster" is typical, which as far as I understand, makes the IP addresses of nodes private, which makes you require a LoadBalancer k8s service object to gain a external ip for access to the cluster. IP-Alias is required for private clusters.
```
--enable-private-nodes
--enable-ip-alias
```

I think going with private nodes will require you to enable a Google Cloud NAT though. This is a very simply step that have for me worked without any issue. This is needed because if the GKE nodes doesn't get a public IP, what IP should outgoing traffic from pods on them get? Well, that is what the Cloud NAT will provide - it could be a single reserved IP, or for example three different IPs.

---

Perhaps this flag is required, perhaps its set by default already? `--no-enable-basic-auth`

---

Do you want to have calico installed which will enforce `networkpolicy` resources in kubernetes? `--enable-network-policy`. This will add pods on each node in the cluster and hog some compute resources, i think perhaps 0.1 CPU / node.

---

I'm not sure if we currently use any vertical pod autoscaler, which would restart pods with higher/lower resource requests based on their usage. I'm not sure if enabling this will hog some resources and how much though.